internal/vsp/vsp.go

```diff
@@ -147,9 +147,14 @@ func (c *Client) ProcessManagedTickets(ctx context.Context) error {
 	err := c.wallet.ForUnspentUnexpiredTickets(ctx, func(hash *chainhash.Hash) error {
 		// We only want to process tickets that haven't been confirmed yet.
 		confirmed, err := c.wallet.IsVSPTicketConfirmed(ctx, hash)
-		if err != nil && !errors.Is(err, errors.NotExist) {
-			c.log.Error(err)
-			return nil
+		if err != nil {
+			// NotExist error indicates unmanaged tickets. Skip.
+			if errors.Is(err, errors.NotExist) {
+				return nil
+			}
+
+			// Any other errors should propagate upwards.
+			return err
```

Comment:
I'm unsure about this change.  The non-nil non-NotExist errors are not being unhandled; they are still logged.

If we return an error to the caller (ForUnspentUnexpiredTickets), now it will error out earlier and not loop over all of the unexpired tickets, which doesn't seem to be what we would want to do even if we did propagate errors up to the caller of ProcessManagedTickets.

Reply:
collecting the errors here and then doing a final return of them with [errors.Join](https://pkg.go.dev/errors#Join) seems like the best way to handle this, but we'll need to wait until go 1.21 at least, and drop support for go <=1.19, because that was introduced in go 1.20.

Reply:
I've pushed a commit which approximates the errors.Join behaviour

Reply:
makes sense, i like it.

i don't like your errors.Join replacement though, since it breaks error detection when using errors.As.  We should either use the real errors.Join here (wrapping it under our own errors package) or just yoink the stdlib code directly into our errors.

Reply:
#2261

Reply:
Please review #2266 before proceeding with this PR, then I will rebase this one.